### PR TITLE
Update e2e storage tests for Windows cluster

### PR DESCRIPTION
**Pull request by jingxu97**
This PR fixes a few things for e2e storage suite to run on Windows
cluster.
1. increaes timeout due to longer pod startup time for windows
2. Only set SELinuxOptions or fsGroup if os is not windows
3. Add VolumeSnapshot delete policy for windows

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by jingxu97** [comment]
cc @msau42

File: test/e2e/framework/pod/wait.go [hunk 1]
```diff
@@ -51,7 +51,7 @@ const (
 	podScheduledBeforeTimeout = podListTimeout + (20 * time.Second)
 
 	// podStartTimeout is how long to wait for the pod to be started.
-	podStartTimeout = 5 * time.Minute
+	podStartTimeout = 8 * time.Minute
```
Comment (msau42) [inline test/e2e/framework/pod/wait.go:54]: my only concern is that this changes the timeout for all tests. Is there a way we can have a separate timeout for windows?

Comment (jingxu97) [reply test/e2e/framework/pod/wait.go]: it is possible to do it, I think. But maybe just leave it 5 mins for now and see how flaky the tests will be and adjust later.

**Comment by liggitt** [comment]
verify is failing on #93623, wait until that is resolved to retest

/hold

**Pull request merged**